build.cake

```diff
@@ -0,0 +1,497 @@
+	if(BuildSystem.IsRunningOnAppVeyor)
+	{
+		Information("Upload artifact to AppVeyor...");
+		BuildSystem.AppVeyor.UploadArtifact(_zipFilePath);
+		var msiFile = @".\GitTfs.Setup\GitTfs.Setup.msi";
+		if(FileExists(msiFile))
+			BuildSystem.AppVeyor.UploadArtifact(msiFile);
+	}
+});
+
+void DisplayAuthTokenErrorMessage()
+{
+	var errorMessage = @"Please create a file 'PersonalTokens.config' containing your authentication tokens
+See the file 'PersonalTokens.config.example' for the format and content.";
+
+	throw new Exception(errorMessage);
+}
+
+string ReadToken(string tokenKey, string tokenRegexFormat = null)
+{
+	var authTargetsFile = "PersonalTokens.config";
+
+	Information("Reading token..." + tokenKey);
+
+	if(!FileExists(authTargetsFile))
+		DisplayAuthTokenErrorMessage();
+
+	var personalToken = System.IO.File.ReadAllLines(authTargetsFile).FirstOrDefault(l => l.StartsWith(tokenKey + "="));
+	if(personalToken == null)
+		DisplayAuthTokenErrorMessage();
+
+	personalToken = personalToken.Trim();
+	personalToken = personalToken.Substring(tokenKey.Length+1,personalToken.Length-tokenKey.Length-1);
+	if(tokenRegexFormat == null)
+		return personalToken;
+	
+	var regexToken = new System.Text.RegularExpressions.Regex(tokenRegexFormat);
+	if(!regexToken.IsMatch(personalToken))
+		DisplayAuthTokenErrorMessage();
+	return personalToken;
+}
+
+string GetChocolateyToken()
+{
+	var chocolateyToken = Argument("chocolateyToken", "");
+	if(!string.IsNullOrEmpty(chocolateyToken))
+		return chocolateyToken;
+	return ReadToken("Chocolatey");
+}
+
+string GetGithubUserAccount()
+{
+	var gitHubOAuthToken = Argument("gitHubUserAccount", "");
+	if(!string.IsNullOrEmpty(gitHubOAuthToken))
+		return gitHubOAuthToken;
+	return ReadToken("GitHubUserAccount");
+}
+
+string GetGithubAuthToken()
+{
+	var gitHubOAuthToken = Argument("gitHubToken", "");
+	if(!string.IsNullOrEmpty(gitHubOAuthToken))
+		return gitHubOAuthToken;
+	return ReadToken("GitHub", "^[0-9a-f]{40}$");
+}
+
+string ReadReleaseNotes()
+{
+	if(!FileExists(ReleaseNotesPath))
+		return string.Empty;
+	return System.IO.File.ReadAllText(ReleaseNotesPath);
+}
+
+Octokit.GitHubClient GetGithubClient()
+{
+	var githubToken = GetGithubAuthToken();
+	var client = new Octokit.GitHubClient(new Octokit.ProductHeaderValue("git-tfs-releasing"));
+	var tokenAuth = new Octokit.Credentials(githubToken);
+	client.Credentials = tokenAuth;
+	return client;
+}
+
+Task("TriggerRelease").Description("Trigger a release from the AppVeyor build server")
+	.Does(() =>
+{
+	var httpClient = new System.Net.Http.HttpClient();
+	//AppVeyor build data to trigger the git-tfs build + parameters passed to the release build 
+	var content = @"{
+accountName: 'pmiossec',
+projectSlug: 'git-tfs-v2qcm',
+branch: 'master',
+environmentVariables: {
+ target: 'AppVeyorRelease',
+ chocolateyToken: '"+ GetChocolateyToken() + @"',
+ gitHubUserAccount: '"+ GetGithubUserAccount() + @"',
+ gitHubToken: '" + GetGithubAuthToken() + @"'
```

Comment:
>> As a first step, before making appveyor do the build, could the release build happen on a local workstation for now?

> I built the script to be able to run it localy or through the build. If the token is not passed to the script (server way to do), it looks for tokens in the file.


Since it's possible to build a release locally, I'm not too concerned about the security of tokens, for now. 

>> Also, if this is going to be the official release process, can we move the appveyor build so that it's also called "git-tfs/git-tfs" there?

> Could you be more specific about that?

> Do you want to create a git-tfs account where we will create the "official" git-tfs build? It should be quite simple because ALL the config is in the yaml file. And perhaps that's better because that way we could share the passwords and login both of us...

This is more what I was asking for. It's not that big of a deal, and can wait. It'd be nice to do this at some point as we get more confident in the appveyor-based releases.